### PR TITLE
Wait until success to update coupon use

### PR DESCRIPTION
**Pull request by johnjeng**
Fixes case when failed subscriptions would count against the user's coupon count.

@DrXyzzy can you test this?

### CODE REVIEW

File: src/smc-hub/client.coffee [hunk 1]
```diff
@@ -2239,6 +2239,15 @@ class exports.Client extends EventEmitter
...
+                    dbg("Update the subscription.")
+                    changes =
+                        quantity : mesg.quantity
+                        plan     : mesg.plan
+                        coupon   : mesg.coupon_id
+                    @_stripe.customers.updateSubscription(customer_id, subscription_id, changes, cb)
+                (cb) =>
+                    @database.stripe_update_customer(account_id : @account_id, stripe : @_stripe, customer_id : customer_id, cb: cb)
                 (cb) =>
                     if mesg.coupon_id
```
Comment (williamstein) [inline src/smc-hub/client.coffee]: There is no cb called at all if mesg.coupon_id isn't true!  Oops.  Make the code

```
if not mesg.coupon_id
    cb()
... then the rest not indented
```

Comment (johnjeng) [reply src/smc-hub/client.coffee]: Fixed. Weird how this change is being recorded.

Comment (williamstein) [reply src/smc-hub/client.coffee]: You forgot the return after the cb(), which is even worse :-)

**Comment by williamstein** [comment]
i'll just merge and fix myself.

**Pull request merged**